### PR TITLE
feat(setting): Disable minimum window size #834

### PR DESCRIPTION
**Pull request by AutumnVN**
owo

### CODE REVIEW

File: src/components/VencordSettings/VencordTab.tsx [hunk 1]
```diff
@@ -83,6 +83,11 @@ function VencordSettings() {
                 key: "winCtrlQ",
                 title: "Register Ctrl+Q as shortcut to close Discord (Alternative to Alt+F4)",
                 note: "Requires a full restart"
+            },
+            !IS_WEB && {
```
Comment (Vendicated) [inline src/components/VencordSettings/VencordTab.tsx:87]:
```suggestion
            IS_DISCORD_DESKTOP && {
```

File: src/main/patcher.ts [hunk 2]
```diff
@@ -107,9 +107,16 @@ if (!IS_VANILLA) {
     };
 
     // Patch appSettings to force enable devtools
```
Comment (Vendicated) [inline src/main/patcher.ts:109]:
```suggestion
    // Patch appSettings to force enable devtools and optionally disable min size
```

File: src/main/patcher.ts [hunk 3]
```diff
@@ -107,9 +107,16 @@ if (!IS_VANILLA) {
...
-    onceDefined(global, "appSettings", s =>
-        s.set("DANGEROUS_ENABLE_DEVTOOLS_ONLY_ENABLE_IF_YOU_KNOW_WHAT_YOURE_DOING", true)
-    );
+    onceDefined(global, "appSettings", s => {
+        s.set("DANGEROUS_ENABLE_DEVTOOLS_ONLY_ENABLE_IF_YOU_KNOW_WHAT_YOURE_DOING", true);
+        if (settings.disableMinSize) {
+            s.set("MIN_WIDTH", 0);
+            s.set("MIN_HEIGHT", 0);
+        } else {
+            s.set("MIN_WIDTH", 940);
```
Comment (Vendicated) [inline src/main/patcher.ts:116]: I think s.delete("MIN_WIDTH") might be better? I think that's a thing, tho it might also be .del()

Comment (AutumnVN) [reply src/main/patcher.ts:116]: it says both not exist

**Review by Vendicated — APPROVED** [review]

**Pull request merged**